driver/normalizer/strconv.go

```diff
@@ -1,18 +1,44 @@
 package normalizer
 
 import (
+	"fmt"
 	"strconv"
+	"strings"
 	"unicode/utf8"
 )
 
 // Functions below are copied from strconv.Unquote and strconv.Quote.
 // Original functions are unable to escape/unescape values containing
 // multiple characters since in Go single quotes represent a rune literal
+// https://github.com/golang/go/blob/65a54aef5bedbf8035a465d12ad54783fb81e957/src/strconv/quote.go#L360
 
 // unquoteSingle is the same as strconv.Unquote, but uses ' as a quote.
 func unquoteSingle(s string) (string, error) {
+	n := len(s)
+	if n < 2 {
+		return "", fmt.Errorf("%+q is not a quoted string", s)
+	}
+	quote := s[0]
+	if quote != s[n-1] {
+		return "", fmt.Errorf("%+q does not begin and end with a quote", s)
+	}
+	s = s[1 : len(s)-1]
+
+	if contains(s, '\n') {
+		return "", fmt.Errorf("%+q contains EOL", s)
+	}
+
+	// Is it trivial? Avoid allocation.
+	if !contains(s, '\\') && !contains(s, quote) {
+		r, size := utf8.DecodeRuneInString(s)
+		if size == len(s) && (r != utf8.RuneError || size != 1) {
+			return s, nil
+		}
+	}
+	s = replaceEscapedMaybe(s, "\\0", "\x00") // treatment of special JS escape seq
+
 	var runeTmp [utf8.UTFMax]byte
-	buf := make([]byte, 0, 3*len(s)/2)
+	buf := make([]byte, 0, 3*len(s)/2) // Try to avoid more allocations.
```

Comment:
I'm not convinced this is worth the cost, but if we do want to micro-optimize here, I'd assert we should not guess but should instead count the escapes on the way past:

```go
nesc := strings.Count(s, "\\")
if nesc == 0 && !strings.Contains(s, quote) {
   …
}
buf := make([]byte, 0, len(s)+3*nesc/2
```

or words to that effect. Otherwise we are allocating a potentially large buffer to no real purpose given that escapes should be expected to be rare.

Reply:
This and above is direct copy from stdlib, no changes :/ 

Just noticed that https://github.com/bblfsh/javascript-driver/pull/64#discussion_r264401155 was somehow lost, so answering here

So far, in my experience maintaining this code - it was _SO much easier_ when the copy is as close as possible to original for 2 reasons:
 - pieces that are copy are clearly identifiable as possible
 - we potentially would need to keep this in sync and eventually just to move to the better version of this in strdlib (e.g grpc has to do the same, as they have special requirements on unquoting for compatibility with c)

So I would suggest keeping it that way for those reason.

In any case - this PR does not change that code/logic, only adds a comment copied from upstream, so I do not think it's that useful to increase the scope of this work by replacing it with some custom logic (given that it delays a release of the driver with a lot of bugs fixed).

But please, let me know if you have strong opinion on that, and we can keep improving this.

Reply:
That all makes sense. I don't feel strongly that we have to change it.